pkg/boot/syslinux/syslinux.go

```diff
@@ -370,15 +379,15 @@ func (c *parser) append(ctx context.Context, config string) error {
 				// read
 				// https://wiki.syslinux.org/wiki/index.php?title=Directives/append
 				// Multiple initrds are comma-separated
-				var initrds []io.ReaderAt
+				var initrds []io.Reader
 				for _, f := range strings.Split(arg, ",") {
-					i, err := c.getFile(f)
+					i, err := c.getFileWithoutCache(f)
 					if err != nil {
 						return err
 					}
 					initrds = append(initrds, i)
 				}
-				e.Initrd = boot.CatInitrds(initrds...)
+				e.Initrd = boot.CatInitrdsWithFileCache(initrds...)
```

Comment:
any reason this is changed to `CatInitrdsWithFileCache ` ? -- why cache is needed ? or behavior is same jsut function name changed ?

Reply:
Spoke offline but summarizing here: since `getFileWithoutCache` returns io.Reader, we'll need to use `CatInitrdsWithFileCache`